src/VisualStudio/Core/Def/Implementation/Venus/ContainedLanguage.cs

```diff
@@ -133,7 +132,13 @@ internal partial class ContainedLanguage<TPackage, TLanguageService>
                 }
             }
 
-            var documentId = this.Project.AddSourceTextContainer(SubjectBuffer.AsTextContainer(), filePath);
+            var documentId = this.Project.AddSourceTextContainer(
+                SubjectBuffer.AsTextContainer(), filePath,
+                sourceCodeKind: SourceCodeKind.Regular, folders: default,
+                documentServiceProvider: ContainedDocument.DocumentServiceProvider.Instace);
+
+            // all contained document is opened by default
+            Workspace.OpenDocument(documentId);
```

Comment:
This isn't technically safe, since if AddSourceTextContainer was batched, it's still not here yet. I think we're best off just adding the flag and shuttling around the value somewhere.

Reply:
sure.